sdk/spring/azure-spring-integration-core/src/main/java/com/azure/spring/integration/core/api/reactor/DefaultMessageHandler.java

```diff
@@ -161,17 +161,23 @@ private String toDestination(Message<?> message) {
 
     private PartitionSupplier toPartitionSupplier(Message<?> message) {
         PartitionSupplier partitionSupplier = new PartitionSupplier();
-        String partitionKey = message.getHeaders().get(AzureHeaders.PARTITION_KEY, String.class);
-        if (!StringUtils.hasText(partitionKey) && this.partitionKeyExpression != null) {
-            partitionKey = this.partitionKeyExpression.getValue(this.evaluationContext, message, String.class);
+        // Priority setting partitionId
+        String partitionId = message.getHeaders().get(AzureHeaders.PARTITION_ID, String.class);
+        if (!StringUtils.hasText(partitionId) && message.getHeaders().containsKey(AzureHeaders.PARTITION_OVERRIDE)) {
+            partitionId = String.valueOf(message.getHeaders().get(AzureHeaders.PARTITION_OVERRIDE));
+        } else if (!StringUtils.hasText(partitionId) && message.getHeaders().containsKey(AzureHeaders.PARTITION_HEADER)) {
+            partitionId = String.valueOf(message.getHeaders().get(AzureHeaders.PARTITION_HEADER, Integer.class));
         }
```

Comment:
please try to refactor the code to make it easier to read

Reply:
Refactored.